redfish-core/lib/account_service.hpp

```diff
@@ -1541,6 +1541,138 @@ inline void updateUserProperties(
     });
 }
 
+inline void uploadACF(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                      const std::vector<uint8_t>& decodedAcf)
+{
+    crow::connections::systemBus->async_method_call(
+        [asyncResp](const boost::system::error_code ec,
+                    sdbusplus::message::message& m,
+                    const std::tuple<std::vector<uint8_t>, bool, std::string>&
+                        messageFDbus) {
+        if (ec)
+        {
+            BMCWEB_LOG_ERROR("DBUS response error: {}", ec);
+            if (strcmp(m.get_error()->name, "xyz.openbmc_project.Certs.Error."
+                                            "InvalidCertificate") == 0)
+            {
+                redfish::messages::invalidUpload(
+                    asyncResp->res,
+                    "/redfish/v1/AccountService/Accounts/service",
+                    "Invalid Certificate");
+            }
+            else
+            {
+                messages::internalError(asyncResp->res);
+            }
+            return;
+        }
+        getAcfProperties(asyncResp, messageFDbus);
+        },
+        "xyz.openbmc_project.Certs.ACF."
+        "Manager",
+        "/xyz/openbmc_project/certs/ACF", "xyz.openbmc_project.Certs.ACF",
+        "InstallACF", decodedAcf);
+}
+
+inline void triggerUnauthenticatedACFUpload(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    std::optional<nlohmann::json> oem)
+{
+    std::optional<nlohmann::json> ibm;
+    if (!redfish::json_util::readJson(*oem, asyncResp->res, "IBM", ibm))
+    {
+        BMCWEB_LOG_ERROR ("Illegal Property ");
+        messages::propertyMissing(asyncResp->res, "IBM");
+        return;
+    }
+
+    if (ibm)
```

Comment:
I think this deeper nested code block can be simplified by doing

```
   std::optional<nlohmann::json> acf;
   std::vector<uint8_t> decodedAcf;
   std::optional<std::string> acfFile;

    if (ibm)
    {
        if (!redfish::json_util::readJson(*ibm, asyncResp->res, "ACF", acf))
        {
            BMCWEB_LOG_ERROR ("Illegal Property ");
            messages::propertyMissing(asyncResp->res, "ACF");
            return;
        }
   }

  if(acf)
  {
     ... acf & acfFile reading...
   }

   std::string sDecodedAcf;
   if (!crow::utility::base64Decode(*acfFile, sDecodedAcf))
   {
                    BMCWEB_LOG_ERROR ("base64 decode failure ");
                    messages::internalError(asyncResp->res);
                    return;
   }
  
```

Reply:
Ack.
For variable declaration I think we should declare it need basis, so it is placed just before it's use.

Reply:
I think it is about balancing of the readability...

I meant it can be short-cut by returning it s

```
  if(!acf)
  {
      return;
 }

  std::vector<uint8_t> decodedAcf;
 std::optional<std::string> acfFile{};
   ... acf & acfFile reading...
  if(failed) {
    return;
 }
...

````